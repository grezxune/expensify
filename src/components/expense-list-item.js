import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt }) => {
    return (
        <div>
            <Link to={`/edit/${id}`}>{description}</Link>
            <p>{amount / 100} - {createdAt}</p>
        </div>
    );
};

export default ExpenseListItem;