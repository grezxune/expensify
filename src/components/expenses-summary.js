import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import getExpensesTotal from '../selectors/expenses-total';
import getVisibleExpenses from '../selectors/expenses';

export const ExpensesSummary = ({ expensesCount, expensesTotal }) => {
    const expenseWord = expensesCount === 1 ? '' : 's';
    const expenseTotalFormatted = numeral(expensesTotal / 100).format('$0,0.00');
    return (
        <div>
            <h1>Viewing {expensesCount} expense{expenseWord} totalling {expenseTotalFormatted}</h1>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);

    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: getExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);