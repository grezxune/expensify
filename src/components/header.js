import React from 'react';
import {NavLink} from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

const Header = (props) => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/dashboard" activeClassName="is-active">Home</NavLink>
        <NavLink to="/create" activeClassName="is-active">Create</NavLink>
        <button onClick={props.startLogout}>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

const mappedComponent = connect(undefined, mapDispatchToProps)(Header);

export { Header, mappedComponent as default }