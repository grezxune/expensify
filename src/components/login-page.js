import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

const LoginPage = (props) => {
    return (
        <div>
            <button onClick={props.startLogin}>login</button>
        </div>
    );
};

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

const mappedComponent = connect(undefined, mapDispatchToProps)(LoginPage);

export { LoginPage, mappedComponent as default };
