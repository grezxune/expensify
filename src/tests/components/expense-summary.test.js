import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/expenses-summary';
import expenses from '../fixtures/expenses';

test('should render ExpenseSummary correctly with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary correctly with two expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={2} expensesTotal={2350000000} />);
    expect(wrapper).toMatchSnapshot();
});