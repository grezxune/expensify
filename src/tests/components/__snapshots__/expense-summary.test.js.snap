// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary correctly with one expense 1`] = `
<div>
  <p>
    Viewing 
    1
     expense
     totalling 
    $2.35
  </p>
</div>
`;

exports[`should render ExpenseSummary correctly with two expenses 1`] = `
<div>
  <p>
    Viewing 
    2
     expense
    s
     totalling 
    $23,500,000.00
  </p>
</div>
`;
